### PR TITLE
Add rvalue constructor to xmlt

### PR DESCRIPTION
**Pull request by tautschnig**
This enables RAII and a more concise notation.

- [x] Each commit message has a non-empty body, explaining why the change was made.
- [ ] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- n/a The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- n/a My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- n/a White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

File: src/util/xml.h [hunk 1]
```diff
@@ -27,6 +27,13 @@ class xmlt
   typedef std::list<xmlt> elementst;
   typedef std::map<std::string, std::string> attributest;
 
+  xmlt(std::string _name, attributest &&_attributes, elementst &&_elements)
+    : name(std::move(_name)),
```
Comment (kroening) [inline src/util/xml.h:31]: Should _name get a `&&`?

Comment (tautschnig) [reply src/util/xml.h:31]: I was wondering how well that would work given the common case is a C string at that point. Or maybe it actually makes no difference?

Comment (kroening) [reply src/util/xml.h:31]: std::string has a lot of optimisation behind it (I've seen instances with copy-on-write, short-string stored in the class itself, etc.); however, disregarding this, yes, it should.

Comment (tautschnig) [reply src/util/xml.h:31]: I've now added the `&&`.

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 61f23ccb5da27b4ecc611c4f60fee7844bdb6bce).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/95336005

File: src/goto-programs/graphml_witness.cpp [hunk 2]
```diff
@@ -302,9 +304,9 @@ void graphml_witnesst::operator()(const goto_tracet &goto_trace)
           val.data="["+(it->cond_value ? cond : neg_cond)+"]";
 
           #if 0
-          xmlt edge2("edge");
-          edge2.set_attribute("source", graphml[from].node_name);
-          edge2.set_attribute("target", graphml[sink].node_name);
+          xmlt edge2("edge", {
```
Comment (smowton) [inline src/goto-programs/graphml_witness.cpp:307]: Style

**Review by smowton — APPROVED** [review]

**Pull request merged**